### DISCUSSION TITLE
gitlab-runner fails to connect

### DISCUSSION THREAD

**Discussion by nickveliki** [minio/minio#16477] [Q&A]
## NOTE
All GitHub issues are addressed on a best-effort basis at MinIO's sole discretion. There are no Service Level Agreements (SLA) or Objectives (SLO). Remember our [Code of Conduct](https://github.com/minio/minio/blob/master/code_of_conduct.md) when engaging with MinIO Engineers and the larger community.

For urgent issues (e.g. production down, etc.), subscribe to [SUBNET](https://min.io/pricing?jmp=github) for direct to engineering support.
 

## Expected Behavior

idk it should connect and not timeout?

## Current Behavior
output in gitlab-runner

```
WARNING: Get "http://[Filtered]:9000/gitlab-ci/project/23/devel-2-protected?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=[FILTERED]&X-Amz-Date=20230125T151251Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=[FILTERED] dial tcp [Filtered]:9000: i/o timeout
```

when I open the link it says the key doesn't exist

```
<Error>
<Code>NoSuchKey</Code>
<Message>The specified key does not exist.</Message>
<Key>project/23/devel-2-protected</Key>
<BucketName>gitlab-ci</BucketName>
<Resource>/gitlab-ci/project/23/devel-2-protected</Resource>
<RequestId>173D9782F44E7B8A</RequestId>
<HostId>fa0d4dcc-c6d5-40b2-952c-f85380945771</HostId>
</Error>
```

## Steps to Reproduce (for bugs)

in gitlab-ci runner config.toml

[runners.cache]
    Type = "s3"
    Shared = true
    MaxUploadedArchiveSize = 0
    [runners.cache.s3]
      ServerAddress = "[Filtered]:9000"
      AccessKey = "[Filtered]"
      SecretKey = "[filtered]"
      BucketName = "gitlab-ci"
      Insecure = true

then run a pipeline with the thus configured runner

## Context
I want to share chache between concurrently running test containers, and the ability to easily discard the cache without breaking the runners (yes, that can happen if oyur cache storage backend is in /var/lib/docker/volumes)

## Regression

No
## Your Environment

quay/minio/minio:latest at time of posting this issue
minio version RELEASE.2023-01-25T00-19-54Z (commit-id=50d58e9b2dbbfbe2f3afa2c16ae701f9c1cf03ec)
Runtime: go1.19.4 linux/amd64
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-2023 MinIO, Inc.

**Comment by harshavardhana** [comment]
@nickveliki how is this a MinIO problem? it looks like you had a transient network issue on your client.

**Reply by nickveliki** [reply]
the client is a docker runner on the same host, the host is listening on port 9000 and 9001 of eth0 interface. Following its own standard route through the gateway, which IS the host, it should be able to connect on the host's eth0:9000. I would rather expect users of minio to contain a larger (relatively) subset of users who know what the problem is than with gitlab. This type of configuration works for other services just fine (TM)

**Reply by RalleKa** [reply]
I agree. This appears not to be a problem with minio, but rather gitlab-runner (or your configuration therefore).

In hopes of being able to help you anyway, try the following (it fixed a similar problem for me):

```
[runners.cache]
Type = "s3"
Shared = true
MaxUploadedArchiveSize = 0
[runners.cache.s3]
ServerAddress = "[Filtered]:9000"
AccessKey = "[Filtered]"
SecretKey = "[filtered]"
BucketName = "gitlab-ci"
Insecure = true
AuthenticationType = "iam" <--------------------------
```
